io-engine/src/grpc/v1/pool.rs

```diff
@@ -57,6 +67,42 @@ impl From<GrowPoolRequest> for FindPoolArgs {
     }
 }
 
+/// Helper routine to extract Encryption params from the Create or Import pool request.
+async fn util_fetch_secret_params(
+    params: &PoolEncryptionParams,
+) -> Result<Option<PoolEncKey>, Status> {
+    let enc_key = match params {
+        PoolEncryptionParams::Create(enc_arg) => {
+            match enc_arg.clone() {
+                // Would have been nice to deduplicate the code if we could call secret_data using a
+                // trait object returned by library. But it doesn't seem to work here because the
+                // library call having generic type in definition isn't allowing that.
+                create_pool_request::Encryption::Data(ckd) => ckd.try_into().ok(),
+                create_pool_request::Encryption::Secret(cks) => {
+                    let secret_params = secret_data(cks.secret.as_str())
+                        .await
+                        .map_err(|e| Status::invalid_argument(e.to_string()))?;
+                    trace!("[create pool] Received encryption params: {secret_params:?}");
+                    Some(secret_params)
+                }
+            }
+        }
+        PoolEncryptionParams::Import(enc_arg) => match enc_arg.clone() {
+            import_pool_request::Encryption::Data(ikd) => ikd.try_into().ok(),
```

Comment:
hmm should these try_into be failed rather than ok'ed?
I might be misunderstanding this, but if we ok() does it mean we end up creating the pool without encryption rather than fail with invalid args?